### PR TITLE
ssl resource: properly raise error when unable to determine if port is enabled

### LINKED ISSUE #1205: resource fail (exception) aborts all tests
### Description

The ssl resource fails if it cannot connect to an external host https://github.com/chef/inspec/blob/master/lib/resources/ssl.rb#L55. This results in a strange behavior, if this is only one tests in a profile, since the full profile execution is aborted:

```
 b inspec exec https://github.com/chris-rock/acme-inspec-profile -t docker://67bcf1bf4019
Cannot determine host for SSL test. Please specify it or use a different target.
```
### InSpec and Platform Version

InSpec 1.0
### Possible Solutions

We should not use fail in the resource or handle resource failures as a control failure.

### PR DESCRIPTION
**Pull request by jquick**
This is related to #1205. This will fix the ssl resource for now until we redo the exceptions. Still looking around the code and need to build some unit tests for the ssl resource.
screen shot 2017-10-03 at 10 38 59 am

My fix here is to move the raise condition till later in the flow, specifically the enabled? method. This lets the raise get caught accordingly without killing the other tests.

### CODE REVIEW

File: test/unit/resources/ssl_test.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# encoding: utf-8
+# author: Christoph Hartmann
+# author: Dominik Richter
```
Comment (arlimus) [inline test/unit/resources/ssl_test.rb]: Feel free to remove our names from the authors list :)

Comment (jquick) [reply test/unit/resources/ssl_test.rb]: Sounds good, just pushed out a change to remove them.

**Review by adamleff — APPROVED** [review]
Thanks, @jquick!

**Review by chris-rock — APPROVED** [review]
Thank you for that improvement @jquick

**Review by arlimus — APPROVED** [review]
Sweet thank you @jquick :D

**Pull request merged**